src/OpenSpaceToolkit/Astrodynamics/NumericalSolver.cpp

```diff
@@ -0,0 +1,418 @@
+    ostk::core::utils::Print::Line(anOutputStream) << "Integration time step:"                << (timeStep_.isDefined() ? timeStep_.toString() : "Undefined") ;
+    ostk::core::utils::Print::Line(anOutputStream) << "Integration relative tolerance:"                << (relativeTolerance_.isDefined() ? relativeTolerance_.toString() : "Undefined") ;
+    ostk::core::utils::Print::Line(anOutputStream) << "Integration absolute tolerance:"                << (absoluteTolerance_.isDefined() ? absoluteTolerance_.toString() : "Undefined") ;
+
+    displayDecorator ? ostk::core::utils::Print::Footer(anOutputStream) : void () ;
+
+}
+
+NumericalSolver::IntegrationLogType  NumericalSolver::getIntegrationLogType ( ) const
+{
+
+    if (!this->isDefined())
+    {
+        throw ostk::core::error::runtime::Undefined("NumericalSolver") ;
+    }
+
+    return integrationLogType_ ;
+
+}
+
+NumericalSolver::IntegrationStepperType  NumericalSolver::getIntegrationStepperType ( ) const
+{
+
+    if (!this->isDefined())
+    {
+        throw ostk::core::error::runtime::Undefined("NumericalSolver") ;
+    }
+
+    return integrationStepperType_ ;
+
+}
+
+Real                            NumericalSolver::getTimeStep                ( ) const
+{
+
+    if (!this->isDefined())
+    {
+        throw ostk::core::error::runtime::Undefined("NumericalSolver") ;
+    }
+
+    return timeStep_ ;
+
+}
+
+Real                            NumericalSolver::getRelativeTolerance       ( ) const
+{
+
+    if (!this->isDefined())
+    {
+        throw ostk::core::error::runtime::Undefined("NumericalSolver") ;
+    }
+
+    return relativeTolerance_ ;
+
+}
+
+Real                            NumericalSolver::getAbsoluteTolerance       ( ) const
+{
+
+    if (!this->isDefined())
+    {
+        throw ostk::core::error::runtime::Undefined("NumericalSolver") ;
+    }
+
+    return absoluteTolerance_ ;
+
+}
+
+NumericalSolver::StateVector    NumericalSolver::integrateStateForDuration  (   const   StateVector&                anInitialStateVector,
+                                                                                const   Duration&                   anIntegrationDuration,
+                                                                                const   SatelliteDynamics::DynamicalEquationFuncCallback& aDynamicsEquationCallback ) const
+{
+
+    NumericalSolver::StateVector aStateVector = anInitialStateVector ;
+
+    double propDurationInSecs ;
+    if ( (anIntegrationDuration.inSeconds()).isZero() ) // If integration duration is zero seconds long, skip integration
+    {
+        return anInitialStateVector ;
+    }
+    else
+    {
+        propDurationInSecs = (double)(anIntegrationDuration.inSeconds()) ;
+    }
+
+    // Ensure integration starts in the correct direction with the initial time step guess
+    const double adjustedTimeStep = timeStep_ * propDurationInSecs / std::abs(propDurationInSecs) ;
+
+    switch (integrationLogType_)
```

Comment:
I think this can be simplified by making a variable for the integrationStepperType_ and then having non-nested case switches, like:
ControlledErrorStepper stepper_type;
switch (integrationStepperType_) {
  case NumericalSolver::IntegrationStepperType::RungeKuttaCashKarp54:
    stepper = NumericalSolver::error_stepper_type_54();
  case NumericalSolver::IntegrationStepperType::RungeKuttaFehlberg78:
    stepper = NumericalSolver::error_stepper_type_78();
}

Reply:
Can you better explain what you mean by this?